torch/_prims/nvfuser_executor.py

```diff
@@ -88,7 +119,32 @@ def run_node(self, node):
                     assert len(node.args) == 2
                     args, kwargs = self.fetch_args_kwargs_from_env(node)
                     args = [args[0], original_shape, args[1]]
-                    return self.call_function(node.target, args, node.kwargs)
+                    return self.call_function(node.target, args, kwargs)
+
+                # copy_to requires special handling of the output tensors
+                elif node.target in [
+                    torch.ops.nvprims.copy_to,
+                    torch.ops.nvprims.copy_to.default,
+                ]:
+                    assert len(node.args) == 2
+                    source_node = node.args[1]
+                    args, kwargs = self.fetch_args_kwargs_from_env(node)
+                    source = args[1]
+                    # If source is a fusion input, we need to place an operation
+                    # before the copy_to so that the copy is actually performed
+                    if _is_node_in_input(gm, source_node):
+                        source = fd.ops.set(source)
+                    result = self.call_function(node.target, [args[0], source], kwargs)
+                    # Check whether the source tensor is an output tensor
+                    # If it is, we need to drop it now from the fusion output
+                    # (it was implicitly added by the copy_to op)
+                    # it will be added back later using correct expected order
+                    if _is_node_in_output(gm, source_node) and not _is_node_in_input(
+                        gm, source_node
```

Comment:
When `_is_node_in_input(gm, source_node) == True` we hit the other path where we create a new temporary `source`, since it's temporary there's no way it could be marked as output in the graph.
Here we only remove from nvFusion outputs `source` that was marked as output inside `aliasOutputToInput` and then is going to be marked as output here:
https://github.com/pytorch/pytorch/blob/8fe12d53c0e5684f89c56d69af0b62656b28cf6c/torch/_prims/nvfuser_executor.py#L172-L175

Reply:
I removed this code, there's no special handling on the Python side anymore.